### PR TITLE
Refactor ecs util with retrier

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

Create an ECS Util with a retrier.

### Motivation

Have more resiliency to connect to the ECS agent

### CODE REVIEW

**Review by JulienBalestra — DISMISSED** [review]
lgtm

File: pkg/util/ec2/ec2.go [hunk 1]
```diff
@@ -83,7 +83,8 @@ func IsDefaultHostname(hostname string) bool {
 
 // HostnameProvider gets the hostname
 func HostnameProvider(hostName string) (string, error) {
-	if ecs.IsInstance() || IsDefaultHostname(hostName) {
+	_, err := ecs.GetUtil()
+	if err != nil || IsDefaultHostname(hostName) {
```
Comment (hkaj) [inline pkg/util/ec2/ec2.go]: don't we want `err == nil` here?

Comment (mfpierre) [reply pkg/util/ec2/ec2.go]: good catch, bad error handling habit 😑

**Review by hkaj — APPROVED** [review]
lgtm

**Pull request merged**